taskvine/src/bindings/python3/taskvine.binding.py

```diff
@@ -1574,25 +1580,104 @@ def submit(self, task):
+        library_env_path = f"{library_cache_path}/library_env.tar.gz"
+
+        # library cache folder doesn't exist, create it
+        if not os.path.exists("vine-library-cache"):
+            os.mkdir("vine-library-cache")
+        # if the library code and environment exist, move on to creating the Library Task
+        if os.path.isfile(library_code_path) and os.path.isfile(library_env_path):
+            pass
+        else:
+            # check if the cache entry exits, and create it if it doesn't exist 
+            if not os.path.exists(library_cache_path):
+                os.mkdir(library_cache_path)
+            print("No cached Library code and environment found, regenerating...")
+            # create library code and environment
+            package_serverize.serverize_library_from_code(library_cache_path, function_list, name)
+            # enable correct permissions for library code
+            os.chmod(library_code_path, 0o775)
+        
+        # create Task to execute the Library
+        t = LibraryTask("./poncho_package_run -e package python ./library_code.py", name)
+        # declare the environment
+        f = self.declare_poncho(self.declare_file(library_env_path))
+        t.add_input(f, "package")
+        # declare the library code as an input
+        f = self.declare_file(library_code_path)
+        t.add_input(f, "library_code.py")
+        # add poncho_package_run to be able to execute the Library
+        f = self.declare_file(shutil.which('poncho_package_run'))
+        t.add_input(f, "poncho_package_run")
+        return t
+
+    # Turn Library code created with poncho_package_serverize into a Library Task
+    #
+    #
+    # @param self            Reference to the current manager object.
+    # @param library_path    Path of the file which contains the library code - the output of poncho_package_serverize
+    # @param env_path        Path of the environment file which contains the environment capable of running the Library
+    def create_library_from_serverized_files(self, name, library_path, env_path):
+        if poncho_available == False:
+            raise ModuleNotFoundError("The poncho module is not available. Cannot create library.")
+        t = LibraryTask(f"./poncho_package_run -e package python ./library_code.py", name)
+        f = self.declare_poncho(self.declare_file(env_path))
+        t.add_input(f, "package")
+        f = self.declare_file(library_path)
+        t.add_input(f, "library_code.py")
+        f = self.declare_file(shutil.which("poncho_package_run"))
+        t.add_input(f, "poncho_package_run")
+        return t
+
+    # Create a Library task from arbitrary inputs
+    #
+    #
+    # @param self            Reference to the current manager object.
+    # @param executable_path Path of the file which contains the library executable
+    # @param name            Name of the library to be created
+    def create_library_from_command(self, executable_path, name):
```

Comment:
Add an argument for an optional environment.  If not None, add the environment to the the task as above.